Makefile

```diff
@@ -451,6 +451,7 @@ $(eval $(call assert_boolean,ERROR_DEPRECATED))
 $(eval $(call assert_boolean,GENERATE_COT))
 $(eval $(call assert_boolean,HW_ASSISTED_COHERENCY))
 $(eval $(call assert_boolean,LOAD_IMAGE_V2))
+$(eval $(call assert_boolean,MULTI_CONSOLE_API))
```

Comment:
Sorry, what I meant to say was the new API should be enabled non-configurably. So all the code under `MULTI_CONSOLE_API = true` should always be compiled in, and the other under `MULTI_CONSOLE_API = false` should only be compiled in when `ERROR_DEPRECATED = false`.

For example, in console.S, instead of:

```
#if MULTI_CONSOLE_API
#include "multi_console.S"
#else
#include "deprecated_console.S"
#endif
```

Have this:

```
// put all content of multi_console.S inline here
#if !ERROR_DEPRECATED
#include "deprecated_console.S"
#endif
```

Then the deprecated APIs in deprecated_console.S should be implemented using the new APIs instead. Similarly in console drivers that support the deprecated APIs, the deprecated APIs should be implemented using the new APIs (currently it's the other way around).

Reply:
Oh yes, I remember now. If we can't make the old APIs co-exist easily, then I agree that keeping this under a separate macro is better than using ERROR_DEPRECATED directly. At the point we remove deprecated APIs (i.e. a future v2 release), I expect we can remove MULTI_CONSOLE_API too and just keep the new code.

So sorry for the confusion and ignore my previous comments here.